BlueprintUICommonControls/Sources/ScrollView.swift

```diff
@@ -485,3 +499,21 @@ extension ScrollerWrapperView : KeyboardObserverDelegate {
     }
 }
 
+
+extension ScrollView {
+
+    final public class Trigger {
+
+        var action: () -> Void
+
+        public init() {
+            action = { }
+        }
+
+        public func fire() {
```

Comment:
I wonder if we should:
a) Take in an `animated` param here
b) Add an `enum` that could control various scrolling options: Scroll to top, or bottom, etc.

Reply:
> I wonder if we should:
> a) Take in an `animated` param here
> b) Add an `enum` that could control various scrolling options: Scroll to top, or bottom, etc.

Yeah, I agree. I could see something like this being real handy:

```swift
public struct ContentOffset {
    public struct ScrollingContext {
        public var contentSize: CGSize
        public var scrollViewBounds: CGRect
        public var contentInsets: UIEdgeInsets
    }

    public typealias Provider = (ScrollingContext) -> CGPoint

    public static let top = ContentOffset { _ in .zero }
    public static let bottom = ContentOffset { context in
        CGPoint(x: 0, y: context.contentSize.height - context.scrollViewBounds.size.height)
    }

    var provider: Provider

    public init(provider: @escaping Provider) {
        self.provider = provider
    }
}
```

And then instead of just `fire()`, you could call it like:
```swift
scrollTrigger.scroll(to: .top, animated: true)
```